operator/src/main/java/oracle/kubernetes/operator/utils/SelfSignedCertUtils.java

```diff
@@ -0,0 +1,141 @@
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.operator.ContentSigner;
+import org.bouncycastle.operator.OperatorCreationException;
+import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder;
+import org.jetbrains.annotations.NotNull;
+
+import static oracle.kubernetes.operator.helpers.NamespaceHelper.getOperatorNamespace;
+
+/**
+ * Utility class for generating key-pair and self-signed certificates.
+ *
+ */
+public final class SelfSignedCertUtils {
+
+  public static final String INTERNAL_WEBLOGIC_OPERATOR_SVC = "internal-weblogic-operator-svc";
+
+  /**
+   * Generates a key pair using the BouncyCastle lib.
+   *
+   * @return Key pair
+   *
+   * @throws InvalidKeySpecException on generating the key if specs are invalid.
+   * @throws NoSuchAlgorithmException in KeyFactory.getInstance() if algorithm is invalid.
+   */
+  public static KeyPair createKeyPair() throws NoSuchAlgorithmException, InvalidKeySpecException {
+
+    RSAKeyPairGenerator rsaKeyPairGenerator = new RSAKeyPairGenerator();
+
+    rsaKeyPairGenerator.init(new RSAKeyGenerationParameters(BigInteger.valueOf(3), new SecureRandom(), 2048, 80));
+    AsymmetricCipherKeyPair keypair = rsaKeyPairGenerator.generateKeyPair();
+
+    RSAKeyParameters publicKey = (RSAKeyParameters) keypair.getPublic();
+    RSAPrivateCrtKeyParameters privateKey = (RSAPrivateCrtKeyParameters) keypair.getPrivate();
+
+    PublicKey pubKey = KeyFactory.getInstance("RSA").generatePublic(
+            new RSAPublicKeySpec(publicKey.getModulus(), publicKey.getExponent()));
+
+    PrivateKey privKey = KeyFactory.getInstance("RSA").generatePrivate(
+            new RSAPrivateCrtKeySpec(publicKey.getModulus(), publicKey.getExponent(),
+                    privateKey.getExponent(), privateKey.getP(), privateKey.getQ(),
+                    privateKey.getDP(), privateKey.getDQ(), privateKey.getQInv()));
+
+    return new KeyPair(pubKey, privKey);
+  }
+
+  /**
+   * Generates a self signed certificate using the BouncyCastle lib.
+   *
+   * @param keyPair used for signing the certificate with PrivateKey
+   * @param hashAlgorithm Hash function
+   * @param commonName Common Name to be used in the subject dn
+   * @param certificateValidityDays validity period in days of the certificate
+   *
+   * @return self-signed X509Certificate
+   *
+   * @throws OperatorCreationException on creating a key id
+   * @throws CertIOException on building JcaContentSignerBuilder
+   * @throws CertificateException on getting certificate from provider
+   */
+  public static X509Certificate generateCertificate(KeyPair keyPair, String hashAlgorithm, String commonName,
+                                                    int certificateValidityDays)
+          throws OperatorCreationException, CertificateException, CertIOException {
+    Instant now = Instant.now();
+    Date notBefore = Date.from(now);
+    Date notAfter = Date.from(now.plus(Duration.ofDays(certificateValidityDays)));
```

Comment:
Interesting... I guess we need to handle regenerating the certificate within some window of it's becoming invalid.

Reply:
Will handle in a separate JIRA.